### PR TITLE
[v21.1.x] backport: release v21.1.0

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Backports for v21.1.0, add release notes and 

## What was done?
See commits

## How Has This Been Tested?
Been on devnet 

## Breaking Changes
None

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: doc/release-notes/dash/release-notes-21.0.1.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# Dash Core version v21.0.2
```
Comment (knst) [inline doc/release-notes/dash/release-notes-21.0.1.md:1]: the file name is wrong, I believe

File: doc/release-notes.md [hunk 2]
```diff
@@ -66,6 +67,7 @@ debug the release candidates.
 
 These release are considered obsolete. Old release notes can be found here:
 
+- [v21.0.1](https://github.com/dashpay/dash/blob/master/doc/release-notes/dash/release-notes-21.0.1.md) released Aug/1/2024
```
Comment (knst) [inline doc/release-notes.md:70]: 21.0.2

**Review by knst — DISMISSED** [review]
utACK 85c7dc5870701941797bdb4fcaaf55d0a35bfb93

consider fix 21.0.1 -> 21.0.2

**Review by knst — APPROVED** [review]
utACK  8e9dd127c774dbe5a3d50cf446a77c7298a58292

File: doc/release-notes.md [hunk 3]
```diff
@@ -58,6 +57,8 @@ Thanks to everyone who directly contributed to this release:
 
 - Konstantin Akimov
 - PastaPastaPasta
+- UdjinM6
+- ogabrielides
```
Comment (UdjinM6) [inline doc/release-notes.md:61]: should be sorted and use proper names
```suggestion
- Konstantin Akimov
- Odysseas Gabrielides
- PastaPastaPasta
- UdjinM6
```

**Review by UdjinM6 — COMMENTED** [review]
ninja merging stuff today huh 🥷 🙂

File: doc/release-notes.md (hunk 3 above)
Comment (PastaPastaPasta) [reply doc/release-notes.md:61]: Too late 🤷

**Pull request merged**